pkg/postgres/tx.go

```diff
@@ -79,7 +86,13 @@ func (t *Tx) Commit(ctx context.Context) error {
 
 // Rollback wraps pgx.Tx Rollback
 func (t *Tx) Rollback(ctx context.Context) error {
-	defer t.cancelFunc()
+	// Only cancel the context for outer transactions to prevent resource leaks.
```

Comment:
This does not look right to me. The tx function should be called for single layer context(original) too. Did you observe the same function called twice? The cancel function should be support re-entry.

Reply:
Reverted this change